server/src/main/java/com/lunchtime/controllers/EventController.java

```diff
@@ -22,7 +24,43 @@ public EventController(EventService eventService) {
 
     @GetMapping
     public ResponseEntity<?> getAll() {
-        List<Event> result = eventService.findAll(new Date());
+        List<Event> result = eventService.findAll();
+        return new ResponseEntity<>(result, HttpStatus.OK);
+    }
+
+    @GetMapping("/categories/{category}")
+    public ResponseEntity<?> getByCategory(@PathVariable String[] category) {
+        List<Event> result = eventService.findByCategory(category);
+        return new ResponseEntity<>(result, HttpStatus.OK);
+    }
+
+    @GetMapping("date/{date}")
+    public ResponseEntity<?> getByDay(@PathVariable @DateTimeFormat(pattern = "yyyy-MM-dd") Date date) {
+        List<Event> result = eventService.findByDay(date);
+        return new ResponseEntity<>(result, HttpStatus.OK);
+    }
+
+    @GetMapping("dates/from={startDate}&to={endDate}")
```

Comment:
I think better path can be `dates?from={startDate}&to={endDate}`
You can see similar url in google search `https://www.google.com/search?client=firefox-b-d&q=spring+unit+testing`

Reply:
Fixed